test/run_e2e_test.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# This script sources the built ROS2 workspace, installs the necessary Python dependencies,
+# and executes the python end to end tests for the system_metrics_collector package.
+
```

Comment:
You probably want `set -euf -o pipefail` here.

Reply:
There are errors when sourcing setup.bash, so this exits prematurely.

Reply:
That sounds problematic - what types of errors?

Reply:
There are a few unset variables in colcon's `setup.bash` that allow for enabling some types of debug output. You can drop the `-u` and it should be fine

Reply:
I agree with Emeron's recommendation. Let's make sure we explicitly check that the variables we need are set then.